openpype/hosts/maya/plugins/publish/validate_visible_only.py

```diff
@@ -0,0 +1,51 @@
+import pyblish.api
+
+import openpype.api
+from openpype.hosts.maya.api.lib import iter_visible_in_frame_range
+import openpype.hosts.maya.api.action
+
+
+class ValidateAlembicVisibleOnly(pyblish.api.InstancePlugin):
+    """Validates at least a single node is visible in frame range.
+
+    This validation only validates if the `visibleOnly` flag is enabled
+    on the instance - otherwise the validation is skipped.
+
+    """
+    order = openpype.api.ValidateContentsOrder + 0.05
+    label = "Alembic Visible Only"
+    hosts = ["maya"]
+    families = ["pointcache", "animation"]
+    actions = [openpype.hosts.maya.api.action.SelectInvalidAction]
+
+    def process(self, instance):
+
+        if not instance.data.get("visibleOnly", False):
+            self.log.debug("Visible only is disabled. Validation skipped..")
+            return
+
+        invalid = self.get_invalid(instance)
+        if invalid:
+            start, end = self.get_frame_range(instance)
+            raise RuntimeError("No visible nodes found in "
+                               "frame range {}-{}.".format(start, end))
+
+    @classmethod
+    def get_invalid(cls, instance):
+
+        if instance.data["family"] == "animation":
+            # Special behavior to use the nodes in out_SET
+            nodes = instance.data["out_hierarchy"]
```

Comment:
### Warning

It's good to be aware that this logic depends on [THIS Collector plug-in](https://github.com/pypeclub/OpenPype/blob/3bb6d8aea083220ee570b61b9c741e03aa7524c9/openpype/hosts/maya/plugins/publish/collect_animation.py#L56-L57).

The bug/issue is that if the instance in Maya is set to `active=False` on the objectSet/instance directly that that Collector does not run.

There are two solutions to that:
- Have the instance collector NOT collect the active state directly but instead collect as e.g. `__active`. Then have a very later Collector (`CollectorOrder + 0.499`) which sets `instance.data["active"] = instance.data,pop("__active", True)` - so that up to that point all collectors do run as expected.
- Use/implement something like Post collectors like `PYBLISH_QML_POST_COLLECT` see: https://github.com/pyblish/pyblish-qml/pull/356 - that way we can have these collectors that should always run for an active instance in the post collection step after the user toggled states. Upside to this is that collecting will be MUCH faster since many collectors could move to the post collector order.